File: sionyx-kiosk-wpf/src/SionyxKiosk/Models/PurchaseStatus.cs
namespace SionyxKiosk.Models;

/// <summary>
/// Purchase status values stored in Firebase.
/// </summary>
public enum PurchaseStatus
{
    Pending,
    Completed,
    Failed
}

/// <summary>
/// Extension methods for PurchaseStatus - Hebrew labels, colors, Nedarim mapping.
/// </summary>
public static class PurchaseStatusExtensions
{
    /// <summary>Firebase database string value.</summary>
    public static string ToDbValue(this PurchaseStatus status) => status switch
    {
        PurchaseStatus.Pending => "pending",
        PurchaseStatus.Completed => "completed",
        PurchaseStatus.Failed => "failed",
        _ => "pending"
    };

    /// <summary>Hebrew display label for UI.</summary>
    public static string ToHebrewLabel(this PurchaseStatus status) => status switch
    {
        PurchaseStatus.Pending => "ממתין",
        PurchaseStatus.Completed => "הושלם",
        PurchaseStatus.Failed => "נכשל",
        _ => status.ToString()
    };

    /// <summary>Semantic color name for UI badges.</summary>
    public static string ToColorName(this PurchaseStatus status) => status switch
    {
        PurchaseStatus.Pending => "processing",
        PurchaseStatus.Completed => "success",
        PurchaseStatus.Failed => "error",
        _ => "default"
    };

    /// <summary>Whether this status is final (no further transitions).</summary>
    public static bool IsFinal(this PurchaseStatus status) =>
        status is PurchaseStatus.Completed or PurchaseStatus.Failed;

    /// <summary>Parse a database string value to PurchaseStatus.</summary>
    public static PurchaseStatus Parse(string? dbValue) => dbValue?.ToLowerInvariant() switch
    {
        "completed" => PurchaseStatus.Completed,
        "failed" => PurchaseStatus.Failed,
        "pending" => PurchaseStatus.Pending,
        _ => PurchaseStatus.Pending
    };

    /// <summary>Map Nedarim payment gateway response to PurchaseStatus.</summary>
    public static PurchaseStatus FromNedarimStatus(string? nedarimStatus) =>
        nedarimStatus == "Error" ? PurchaseStatus.Failed : PurchaseStatus.Completed;
}

File: sionyx-web/src/constants/purchaseStatus.js
/**
 * Purchase Status Constants
 * Unified constants for purchase status values across the application
 * Database stores English values, UI displays Hebrew labels
 */

// Database values (English) - used in Firebase and all services
export const PURCHASE_STATUS = {
  PENDING: 'pending',
  COMPLETED: 'completed',
  FAILED: 'failed',
};

// Display labels (Hebrew) - used in UI
export const PURCHASE_STATUS_LABELS = {
  [PURCHASE_STATUS.PENDING]: 'ממתין',
  [PURCHASE_STATUS.COMPLETED]: 'הושלם',
  [PURCHASE_STATUS.FAILED]: 'נכשל',
};

// Status colors for UI components
export const PURCHASE_STATUS_COLORS = {
  [PURCHASE_STATUS.PENDING]: 'processing',
  [PURCHASE_STATUS.COMPLETED]: 'success',
  [PURCHASE_STATUS.FAILED]: 'error',
};

// Helper function to get Hebrew label for status
export const getStatusLabel = status => {
  return PURCHASE_STATUS_LABELS[status] || status;
};

// Helper function to get color for status
export const getStatusColor = status => {
  return PURCHASE_STATUS_COLORS[status] || 'default';
};

// Helper function to check if status is final (completed or failed)
export const isFinalStatus = status => {
  return status === PURCHASE_STATUS.COMPLETED || status === PURCHASE_STATUS.FAILED;
};
